Custom theme

ashwinKumar0505 opened the pull request.

ashwinKumar0505 committed:
improvement(custom theme): the updated custom theme can also be viewed in themes in modal

ashwinKumar0505 committed:
fix(custom-theme): changed the interactive box padding from 4 to 1 rem

As we are changing from default theme to custom theme. 4 can be given some other spacing values. So
to make the box constant , I changed to 1rem

ashwinKumar0505 committed:
improvement(redux store): custom theme in the redux store is made persistent

ashwinKumar0505 committed:
improvement(clear theme): the clear option to remove the custom themes is added

ashwinKumar0505 committed:
improvement(custom theme): the custom theme will be shown in code preview and code sandbox

ashwinKumar0505 committed:
changed the dispatch action name

ashwinKumar0505 committed:
improvement(font-family): the font family can now be downloaded and can be used in themes

ashwinKumar0505 committed:
improvement(theme properties): overriding of properties

The properties in the theme will be overridden by the properties in the custom theme.

sanandnarayan commented on src/utils/CombineThemes.ts:
```diff
@@ -0,0 +1,25 @@
+        isValidProperty(element, secondTheme) && secondTheme[element]
+  })
+  return combinedTheme
+}
```
there might be a more performant way to do this. 
May be you can see how this works.
https://www.npmjs.com/package/lodash.merge

ashwinKumar0505 replied:
okay bro. I will look after it

ashwinKumar0505 committed:
improvement(load fonts): changed from @fontface to webfontloader to load fonts

ashwinKumar0505 committed:
improvement(theme overriding): the merge from the lodash is used to override properties of theme

ashwinKumar0505 committed:
fix(webfontloader): added webfontloader

ashwinKumar0505 committed:
improvement(persistence): the loaded font family is made persistent

the loaded font family will be added to the redux store and is made persistent.

ashwinKumar0505 committed:
improvement(load fonts): the loaded fonts can now be deleted

ashwinKumar0505 committed:
fixed some issues

ashwinKumar0505 committed:
fix(font family): the font family is now applied to all the components

At first , only the text and heading gets affected by the font family from the theme. Now all the
components get affected by the font family.

ashwinKumar0505 commented:
The webfontloader does not give the exact error. It just says whether the font is active or inactive. So i think we can not show the exact error (font not present or network problem etc.)

ashwinKumar0505 committed:
correct the difference in file stylespanel
